projects/ngx-translate/core/tests/namespace-translate.service.spec.ts

```diff
@@ -0,0 +1,437 @@
+import { TestBed } from "@angular/core/testing";
+import { Observable, of, zip } from "rxjs";
+import { take, toArray, first } from 'rxjs/operators';
+import { TranslateLoader, TranslateModule, TranslateService, NamespaceTranslateService, TRANSLATION_NAMESPACE } from '../src/public_api';
+import * as flatten from "flat";
+
+let translations: any = { "NAMESPACE": { "TEST": "This is a namespace test" } };
+
+class FakeLoader implements TranslateLoader {
+  getTranslation(lang: string): Observable<any> {
+    return of(translations);
+  }
+}
+
+describe('NamespaceTranslateService', () => {
+
+  describe('getNamespacedKey', () => {
+    let namespaceTranslate: NamespaceTranslateService;
+    let translate: TranslateService;
+
+    beforeEach(() => {
+      TestBed.configureTestingModule({
+        imports: [
+          TranslateModule.forRoot({
+            loader: { provide: TranslateLoader, useClass: FakeLoader }
+          })
+        ],
+        providers: [{ provide: TRANSLATION_NAMESPACE, useValue: "NAMESPACE" }]
```

Comment:
Must be:
`
providers: [
          { provide: TRANSLATION_NAMESPACE, useValue: "NAMESPACE" },
          NamespaceTranslateService
        ]
`